sonic_platform_base/sonic_storage/ssd.py

```diff
@@ -126,15 +171,30 @@ def parse_generic_ssd_info(self):
             if temp_raw == NOT_AVAILABLE:
                 self.temperature = NOT_AVAILABLE
             else:
-                self.temperature = temp_raw.split()[-6]
+                self.temperature = temp_raw.split()[7].split()[0]
 
         self.serial = self._parse_re('Serial Number:\s*(.+?)\n', self.ssd_info)
         self.firmware = self._parse_re('Firmware Version:\s*(.+?)\n', self.ssd_info)
 
+        io_reads_raw = self.parse_id_number(GENERIC_IO_READS_ID)
```

Comment:
The parse_id_number API needs to be fixed. Currently it will match any field as id which may lead to invalid parsing

Reply:
Fixed in latest commit, along with couple more bugs.